features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,588 @@
+      if (tabIndex == "deposit") {
+        if (collAllowance === null || collateralToDeposit === null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < parseFloat(collateralToDeposit);
+      }
+      if (tabIndex == "liquidate") {
+        if (collAllowance === null || finalFee == null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < finalFee;
+      }
+      return true;
+    };
+    const needTokenAllowance = () => {
+      if (tokenAllowance === null || maxTokensToLiquidate === null) return true;
+      if (tokenAllowance === "Infinity") return false;
+      return tokenAllowance < parseFloat(maxTokensToLiquidate);
+    };
+
+    // Liquidation requires both collateral and synthetic to be approved. This
+    // results in a number of different configurations the user could be in depending
+    // on their interaction with this Dapp. Process the text here to keep the
+    // component simple.
+    const liquidationNeedAllowanceText = () => {
+      if (!needCollateralAllowance() && !needTokenAllowance()) {
+        return null;
+      }
+      if (needCollateralAllowance() && !needTokenAllowance()) {
+        return `You will need to sign two transactions, one to approve collateral ${collSymbol} and a second to preform the liquidation.`;
+      }
+      if (!needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign two transactions one to approve synthetic ${tokenSymbol} and a second to preform the liquidation.`;
+      }
+      if (needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign three transactions, one to approve synthetic ${tokenSymbol}, one to approve collateral ${collSymbol} and a third to preform the liquidation.`;
+      }
+    };
+
+    const executeDeposit = async () => {
+      if (!collBalanceTooLow()) {
+        setHash(null);
+        setSuccess(null);
+        setError(null);
+        const collateralToDepositWei = toWei(collateralToDeposit);
+
+        try {
+          if (needCollateralAllowance()) await setMaxCollateralAllowance();
+          const tx = await emp.depositTo(props.selectedSponsor, [
+            collateralToDepositWei,
+          ]);
+          setHash(tx.hash as string);
+          await tx.wait();
+
+          setSuccess(true);
+        } catch (error) {
+          console.error(error);
+          setError(error);
+        }
+      } else {
+        setError(new Error("Please check that you are connected."));
+      }
+    };
+
+    const executeLiquidation = async () => {
+      if (!collBalanceTooLow() && !tokenBalanceTooLow) {
+        setHash(null);
+        setSuccess(null);
+        setError(null);
+        const minCollPerTokenWei = toWei(minCollPerToken);
+        const maxCollPerTokenWei = toWei(maxCollPerToken);
+        const maxTokensToLiquidateWei = toWei(maxTokensToLiquidate);
```

Comment:
Should this be `maxTokensToLiquidateNum`

Reply:
same comment as above, cant cast int to big number needs to be string